### PR TITLE
Add callback for WKWebView's webViewWebContentProcessDidTerminate

### PR DESCRIPTION
**Pull request by calmyournerves**
# Summary

(Porting this over from our PR at https://github.com/CRAlpha/react-native-wkwebview/pull/218)

On iOS, WebView's can get terminated (e.g. running too long or using too many resources, not sure about all the reasons here). After getting terminated, they will just show a blank page, making them unusable.

This PR adds a callback for [webViewWebContentProcessDidTerminate](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455639-webviewwebcontentprocessdidtermi?language=objc), allowing this case to be handled (e.g. reloading the webview).

We've been using the code from https://github.com/CRAlpha/react-native-wkwebview/pull/218 in production for quite some time now and never had any issues so far.

### What's required for testing (prerequisites)?

- An app with a webview
- iOS simulator

### What are the steps to reproduce (after prerequisites)?

1. Open your react-native app in the iOS simulator
2. Navigate to a webview
3. Kill the `com.apple.WebKit.WebContent` process in Activity Monitor
4. Your configured callback should get called

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ❌ (inapplicable)     |

## Checklist

- [ ] I have tested this on a device
- [x] I have tested this in a simulator
- [x] I added the documentation in `docs/Reference.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: src/WebViewTypes.ts [hunk 1]
```diff
@@ -424,6 +425,11 @@ export interface IOSWebViewProps extends WebViewSharedProps {
    * @platform ios
    */
   keyboardDisplayRequiresUserAction?: boolean;
+
+  /**
+   * Function that is invoked when the WebKit WebView content process gets terminated.
+   */
+  onContentProcessDidTerminate: (event: WebViewErrorEvent) => void;
```
Comment (Titozzz) [inline src/WebViewTypes.ts]: I don't think this should be a WebViewErrorEvent, please create another type

Comment (calmyournerves) [reply src/WebViewTypes.ts]: Thanks for looking at this!

I added a WebViewTerminatedEvent type. Not sure if I have done it how it is supposed to be, just let me know.

Comment (Titozzz) [reply src/WebViewTypes.ts]: This is fine for me :)

**Comment by Titozzz** [comment]
Please merge master inside your PR. I had to rename some files in the latest release, sorry :'(.

After that, good to go! Thanks for your contribution

**Comment by calmyournerves** [comment]
Cool, thanks for looking at it! Just merged master :).

**Comment by kefyru** [comment]
Will it be added in next 7.0.6 release? It will resolve my problem: webView blank when app comes from background...

**Comment by kefyru** [comment]
Any news?

**Comment by BenWildeman** [comment]
shouldn't have this been made optional? TS says this is required, but there's zero mention of this in the docs

**Pull request merged**